EventGenerator/src/CaloCalibGun_module.cc

```diff
@@ -0,0 +1,284 @@
+    //int                    _nPipes;
+    double                 _pipeRadius;
+    std::vector<double>    _pipeTorRadius;
+    std::vector<double>    _randomRad;
+    CLHEP::Hep3Vector      _zPipeCenter;
+
+    bool _doHistograms;
+    TTree* _Ntupe;
+    float _genErg;
+    float _genTime;
+    float _genCos;
+    float _genPosX;
+    float _genPosY;
+    float _genPosZ;
+  };
+
+  CaloCalibGun::CaloCalibGun(const Parameters& conf)
+    : EDProducer{conf}
+    , _mean{conf().mean()}
+    , _energy{conf().energy()}
+    , _cosmin{conf().cosmin()}
+    , _cosmax{conf().cosmax()}
+    , _phimin{conf().phimin()}
+    , _phimax{conf().phimax()}
+    , _tmin{conf().tmin()}
+    , _tmax{conf().tmax()}
+    , _engine{createEngine(art::ServiceHandle<SeedService>()->getSeed())}
+    , _randFlat{_engine}
+    , _randPoissonQ{_engine, _mean*1.0}
+    , _randomUnitSphere{_engine, _cosmin, _cosmax, 0, CLHEP::twopi}
+  {
+    produces<mu2e::GenParticleCollection>();
+    produces<mu2e::PrimaryParticle>();
+    produces <MCTrajectoryCollection>();
+
+    if ( _doHistograms )
+    {
+      art::ServiceHandle<art::TFileService> tfs;
+      _Ntupe  = tfs->make<TTree>("calibGun", "calibGun");
+      _Ntupe -> Branch("genErg", &_genErg, "genErg/F");
+      _Ntupe -> Branch("genTime", &_genTime, "genTime/F");
+      _Ntupe -> Branch("genCos", &_genCos, "genCos/F");
+      _Ntupe -> Branch("genPosX", &_genPosX, "genPosX/F");
+      _Ntupe -> Branch("genPosY", &_genPosY, "genPosY/F");
+      _Ntupe -> Branch("genPosZ", &_genPosZ, "genPosZ/F");
+    }
+    if (_mean < 0) throw cet::exception("RANGE") << "CaloCalibGun.mean must be non-negative "<< '\n';
+  }
+
+  //================================================================
+  void CaloCalibGun::produce(art::Event& event) {
+    std::unique_ptr<GenParticleCollection> output(new GenParticleCollection);
+    PrimaryParticle primaryParticles;
+    MCTrajectoryCollection mctc;
+
+    _cal =  &*GeomHandle<mu2e::DiskCalorimeter>();
+    //_nPipes          = _cal->caloInfo().getInt("nPipes");
+    _pipeRadius      = _cal->caloInfo().getDouble("pipeRadius");
+    _pipeTorRadius   = _cal->caloInfo().getVDouble("pipeTorRadius");
+    _randomRad       = _cal->caloInfo().getVDouble("pipeTorRadius");
+    _zPipeCenter     = _cal->disk(1).geomInfo().origin()-CLHEP::Hep3Vector(0,0,_cal->disk(1).geomInfo().size().z()/2.0-_pipeRadius);//disk =1
+    // we normalize to the volume of the pipe (proportional to 2*pi*R if they have all the same radius) to draw a random number from which to generate the photons TODO - is this used?
+    double sumR(0);
+    std::for_each(_randomRad.begin(), _randomRad.end(), [&](double& d) {sumR+=d; d = sumR; });
+    std::for_each(_randomRad.begin(), _randomRad.end(), [&](double& d) {d /= sumR;});
+    
+    unsigned int nGen = _mean;
+    //Define the parameters of the pipes:
+    
+    // angle of large torus in degrees
+    double phi_lbd[5] = {161.34, 149.50, 139.50, 132.07, 125.39};
```

Comment:
This information should be in the geometry service and you should get it from there.  If they are not present in the geometry service, please add them.  When you do that, discuss the plan with main author of the calo section - which I believe is Bertrand. The same comment on all of the other places in which you have embedded geometry numbers.

Reply:
This is the long term plan, this is just a first pass to update the old code

Reply:
I would like you to do it as part of this PR.

Reply:
OK, I will discuss with Bertrand

Reply:
Would this be the correct place: https://github.com/Mu2e/Offline/blob/main/GeometryService/src/DiskCalorimeterMaker.cc ?

Reply:
Yes.  And add the appropriate content to CalorimeterGeom/inc/DiskCalorimeter.hh .  It's a good idea to discuss with Bertrand.

Reply:
This has been done now